### ISSUE TITLE
no-array-index-key not triggered by spread props

### ISSUE THREAD

**Issue by vcanales** [yannickcr/eslint-plugin-react#1625]
There's no warning or error when doing something like this:

```jsx
people.map(({  name, email }), key) => {
  const props = {
    key,
    name,
    email,
  };

  return <Person {...props} />;
};
```

I realize this is a bit of an edge case, but maybe we can find a way to catch these uses as well.

**Comment by ljharb** [comment]
No key here is being set to an array index; we don’t have a rule that checks for this.

I don’t believe it’s possible to get anything but `undefined` from `props.key` anyways, so it’d be the same as not passing it.

**Comment by vcanales** [comment]
Sorry, I goofed the syntax on the example; `key` is not `undefined`, it was meant to be the second parameter on `.map`. When using this in real life scenarios, I don't get an eslint error nor does react complain about my iterables having no `key.`
